### ISSUE TITLE
Carousel controls focus when disabled

### ISSUE THREAD

**Issue by aweell** [Telefonica/mistica-design#2330]
## Description

When the user reach the last item of the carousel the button becomes disabled and loses focus, making a change of context to the user.

Seems a possible failure of https://www.w3.org/WAI/WCAG22/Understanding/focus-order.html

### Initial investigation

https://ux.stackexchange.com/a/103244

Buttons that inform about a level of completion could be included in the focus order to give information to the user that has reached the end of the carousel.

### Conclusion

We should allow some interactive elements in some cases to gain focus even if didsabled, because gaining focus gives information to the user about the progress.

For now we will apply only this approach to the carousel component. But could be a generic property of the button component in the furture.

#### Focusability on disabled controls

https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#focusabilityofdisabledcontrols

We shouldn't change default behaviour of disabled buttons but to allow configure if they need to be focusable when disabled or not instead, the default behaviour should be the actual behaviour (to not be focusable)

## Tasks

- [x] Update specs
- [ ] Android ticket
- [ ] iOS ticket
- [x] [Web ticket](https://jira.tid.es/browse/WEB-2312)
- [x] [UXW ticket](https://jira.tid.es/browse/PRODUCTDSN-3485)
